.github/actions/setup-sentry/action.yml

```diff
@@ -0,0 +1,137 @@
+
+        ### services configuration ###
+        # Note: some backend tests (e.g. tests/sentry/eventstream/kafka/test_consumer.py) will behave
+        # differently if these are set.
+        if [ "$NEED_KAFKA" = "true" ]; then
+          echo "SENTRY_KAFKA_HOSTS=127.0.0.1:9092" >> $GITHUB_ENV
+          echo "SENTRY_ZOOKEEPER_HOSTS=127.0.0.1:2181" >> $GITHUB_ENV
+        fi
+
+    # Setup custom pytest matcher, see https://github.com/actions/setup-node/issues/97
+    - name: Add pytest log matcher
+      shell: bash
+      run: |
+        echo "::remove-matcher owner=pytest::"
+        echo "::add-matcher::.github/pytest.json"
+
+    - name: Install system dependencies
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y --no-install-recommends \
+          libxmlsec1-dev \
+          libmaxminddb-dev
+
+    - name: Set up outputs
+      id: config
+      env:
+        MATRIX_INSTANCE: ${{ matrix.instance }}
+      shell: bash
+      run: |
+        echo "::set-output name=yarn-cache-dir::$(yarn cache dir)"
+        echo "::set-output name=matrix-instance-number::$(($MATRIX_INSTANCE+1))"
+        echo "::set-output name=acceptance-dir::.artifacts/visual-snapshots/acceptance"
+
+    - name: Install pip
+      shell: bash
+      run: |
+        pip install --no-cache-dir --upgrade "pip>=20.0.2"
+
+    - name: Get pip cache dir
+      id: pip-cache
+      shell: bash
+      run: |
+        echo "::set-output name=pip-cache::$(pip cache dir)"
+
+    - name: Install python dependencies
+      shell: bash
+      run: |
+        python setup.py install_egg_info
+        pip install wheel # GitHub Actions does not have `wheel` installed by default (unlike Travis)
+        pip install -U -e ".[dev]"
+
+    - name: Start devservices
+      shell: bash
+      env:
+        NEED_KAFKA: ${{ inputs.kafka }}
+        NEED_SNUBA: ${{ inputs.snuba }}
+      run: |
+        sentry init
+
+        # TODO: Use devservices kafka. See https://github.com/getsentry/sentry/pull/20986#issuecomment-704510570
+        if [ "$NEED_KAFKA" = "true" ]; then
+          docker run \
+            --name sentry_zookeeper \
+            -d --network host \
+            -e ZOOKEEPER_CLIENT_PORT=2181 \
+            confluentinc/cp-zookeeper:4.1.0
+          docker run \
+            --name sentry_kafka \
+            -d --network host \
+            -e KAFKA_ZOOKEEPER_CONNECT=127.0.0.1:2181 \
+            -e KAFKA_LISTENERS=INTERNAL://0.0.0.0:9093,EXTERNAL://0.0.0.0:9092 \
+            -e KAFKA_ADVERTISED_LISTENERS=INTERNAL://127.0.0.1:9093,EXTERNAL://127.0.0.1:9092 \
+            -e KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=INTERNAL:PLAINTEXT,EXTERNAL:PLAINTEXT \
+            -e KAFKA_INTER_BROKER_LISTENER_NAME=INTERNAL \
+            -e KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1 \
+            confluentinc/cp-kafka:5.1.2
+        fi
+
+        if [ "$NEED_SNUBA" = "true" ]; then
+          sentry devservices up postgres redis clickhouse snuba
```

Comment:
Let's only start clickhouse and snuba here. Start postgres and redis right after `sentry init`. Generally almost everything needs those two.

Reply:
Okay, we can't do this until I solve a devservices issue - ignore for now.